src/request/reducers/RequestDetailRequestReducer.ts

```diff
@@ -92,20 +75,37 @@ function correlateMiddlewareMessages(startMessages: IMessageEnvelope<IMiddleware
 function toMap<T, TResult>(values: T[], keySelector: (value: T) => string, valueSelector: (value: T) => TResult): { [key: string]: TResult } {
 
     return _.reduce(
-        values, 
+        values,
         (result: { [key: string]: TResult }, value: T) => {
             result[keySelector(value)] = valueSelector(value);
 
             return result;
-        }, 
+        },
         <{ [key: string]: TResult }>{});
 }
```

Comment:
> The one thing it doesn't do is provide a selector for the value.

Yep, and that's almost always what I need (and do, in this case). There are ways in `lodash` to pair and un-pair with a transform that work, but it's more processing and the TypeScript bindings make it less pleasant to work with.